swap_meet/vendor.py

```diff
@@ -1,2 +1,119 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item) 
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        for index in range(len(self.inventory)):
+            current_item = self.inventory[index]
+            if current_item == item:
+                self.inventory[index] = self.inventory[-1] 
+                self.inventory.pop()
+                break
+
+        return item
+    
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+
+    def swap_items(self, other_vendor, my_item, their_item):
+
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
```

Comment:
👍Nice guard clause approach to ensure the items being swapped are actually present. Looking for invalid conditions and exiting allows our main logic to require less indentation.